Orange/widgets/utils/annotated_data.py

```diff
@@ -117,9 +117,16 @@ def create_groups_table(data, selection,
                         var_name=ANNOTATED_DATA_FEATURE_NAME):
     if data is None:
         return None
-    values = ["G{}".format(i + 1) for i in range(np.max(selection))]
+    max_sel = np.max(selection)
+    values = ["G{}".format(i + 1) for i in range(max_sel)]
     if include_unselected:
-        values.insert(0, "Unselected")
+        # Place Unselected instances in the "last group", so that the group
+        # colors and scatter diagram marker colors will match
+        values.append("Unselected")
+        mask = (selection != 0)
+        selection = selection.copy()
+        selection[mask] = selection[mask] - 1
+        selection[~mask] = selection[~mask] = max_sel
```

Comment:
Why not just subtract 1 from everything and then replace -1 with something else?

Reply:
Because in some cases array's dtype is "uint8". In that case subtracting 1 from 0 results in 255.
I was thinking about changing dtype, but I think the above solution is more robust.